__tests__/pet.test.js

```diff
@@ -0,0 +1,203 @@
+    it('decreases hunger, but not to below 0, when feed is called', () => {
+      const pet = new Pet('Fido');
+      pet.hunger = 2;
+      pet.feed();
+      expect(pet.hunger).toEqual(0);
+    });
+  });
+
+  describe('checkUp', () =>{
+    it('pet asks for a walk if unfit', () =>{
+      const pet = new Pet('Fido');
+      pet.fitness = 1;
+      expect(pet.checkUp()).toBe('I need a walk');
+    });
+    it('pet asks for food if hungry', () =>{
+      const pet = new Pet('Fido');
+      pet.hunger = 6;
+      expect(pet.checkUp()).toBe('I am hungry');
+    });
+    it('pet asks for food and a walk if both unfit and hungry', () =>{
+      const pet = new Pet('Fido');
+      pet.fitness = 1;
+      pet.hunger = 6;
+      expect(pet.checkUp()).toBe('I am hungry AND I need a walk');
+    });
+    it('pet declares it is happy if both fed and fit', () =>{
+      const pet = new Pet('Fido');
+      pet.fitness = 4;
+      pet.hunger = 4;
+      expect(pet.checkUp()).toBe('I feel great!');
+    });
+  });
+  
+  describe('deathConditions', () =>{
+    it('pet is dead if fitness is 0 or less', () =>{
+      const pet = new Pet('Fido');
+      pet.fitness = 0;
+      expect(pet.isAlive).toBe(false);
+    });
+    it('pet is dead if hunger is 10 or more', () =>{
+      const pet = new Pet('Fido');
+      pet.hunger = 10;
+      expect(pet.isAlive).toBe(false);
+    });
+    it('pet is dead on the age of 30', () =>{
+      const pet = new Pet('Fido');
+      pet.age = 30;
+      expect(pet.isAlive).toBe(false);
+    });
+    it('pet is still alive if no death conditions are met', () => {
+      const pet = new Pet('Fido');
+      expect(pet.isAlive).toBe(true);
+    })
+  });
+
+describe('deadPet', () => {
+  it('checkUp() function alerts us that the pet is dead (age too high)', () => {
+    const pet = new Pet('Fido');
+    pet.age = 30;
+    expect(pet.checkUp()).toBe('Your pet is no longer alive :(');
+  });
+  it('checkUp() function alerts us that the pet is dead (hunger too high)', () =>{
+    const pet = new Pet('Fido');
+    pet.hunger = 10;
+    expect(pet.checkUp()).toBe('Your pet is no longer alive :(');
+  });
+  it('checkUp() function alerts us that the pet is dead (fitness too low)', () =>{
+    const pet = new Pet('Fido');
+    pet.fitness = 0;
+    expect(pet.checkUp()).toBe('Your pet is no longer alive :(');
+  });
+  it('walk(), feed() and growUp() functions should throw an error if the pet is dead', () => {
+    const pet = new Pet('Fido');
+    pet.age = 30;
+    expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+    expect(() => pet.walk()).toThrow('Your pet is no longer alive :(');
+    expect(() => pet.growUp()).toThrow('Your pet is no longer alive :(');
+  });
+  it('pet should function as normal if still alive', () =>{
+    const pet = new Pet('Fido');
+    expect(pet.checkUp()).toBe('I feel great!');
+  });
+});
+
+describe('adoptedChild', () => {
+  it('Amelia has an adopted child named River', () =>{
```

Comment:
Rather than naming the test after the values being used abstractions are more appropriate, eg adopted childs name is correct